core/src/main/java/org/apache/accumulo/core/clientImpl/ConditionalWriterImpl.java

```diff
@@ -365,8 +371,12 @@ private void reschedule(SendTask task) {
     this.context = context;
     this.auths = config.getAuthorizations();
     this.ve = new VisibilityEvaluator(config.getAuthorizations());
-    this.threadPool = ThreadPools.createScheduledExecutorService(config.getMaxWriteThreads(),
-        this.getClass().getSimpleName(), false);
+    this.threadPool = context.getClientThreadPools().createScheduledExecutorService(
+        config.getMaxWriteThreads(), this.getClass().getSimpleName(), false);
+    this.cleanupThreadPool = context.getClientThreadPools().createFixedThreadPool(1, 3, SECONDS,
+        "Conditional Writer Cleanup Thread", true);
+    this.cleaner = CleanerUtil.shutdownThreadPoolExecutor(cleanupThreadPool, closed,
+        LoggerFactory.getLogger(ConditionalWriterImpl.class));
```

Comment:
This cleanupThreadPool would probably be better left in ClientContext, and shut down when that is closed, rather than store it as an instance member here, where it's only ever used in the close method of this instance. It's very weird to *start* using resources for an object in that object's close method, which should be cleaning up resources. Putting it in ClientContext makes this effectively a singleton again (per Client, not per JVM, though, but that's still better than per-ConditionalWriter) and its own cleanup will be a bit more sane there, I think.

Reply:
Addressed in ec0e7cf